algebra/group_power.lean

```diff
@@ -304,6 +304,20 @@ theorem add_monoid.smul_nonneg [ordered_comm_monoid α] {a : α} (H : 0 ≤ a) :
 | 0     := le_refl _
 | (n+1) := add_nonneg' H (add_monoid.smul_nonneg n)
 
+lemma pow_abs [decidable_linear_ordered_comm_ring α] (a : α) (n) : (abs a)^n = abs (a^n) := 
+begin 
+  induction n with n hi,
+  { simp [monoid.pow] },
+  { simp [*, monoid.pow, abs_mul] }
+end
+
+lemma pow_inv [discrete_field α] (a : α) (n) : (a^n)⁻¹ = (a⁻¹)^n :=
+begin 
+  induction n with n hi, 
+  { simp [monoid.pow] },
+  { simp [*, monoid.pow, mul_inv', mul_comm] }
```

Comment:
`mul_comm` necessary because `mul_inv'` also reverses the order. `(b * a)⁻¹ = a⁻¹ * b⁻¹` Probably `mul_inv'` ought to be changed to `(a * b)⁻¹ = a⁻¹ * b⁻¹`

Reply:
I guess `mul_inv'` is stated in this way as it may work on `division_ring` (where `*` is not commutative). Is it a problem? Does the same `simp [*, monoid.pow, mul_inv', mul_comm]` call also work for the first case?

Reply:
`simp [*, monoid.pow, abs_mul, mul_comm]` does work for the other lemma, but why put `mul_comm` in the brackets when it isn't needed?

Reply:
I want the entire proof in one line! So `by induction n; simp [...]` (the `;` calls the simplifier for all cases after the induction)

Now we lose a little bit of information (which rewrite rules were necessary for the `0` case and which were necessary for the `n+1` case) But in the end we tell the reader than nothing spectacular is happening. It is a simple induction + rewrite proof. These `by induction n; simp [...]` or `by cases h; ...` style proofs are very common. Having them in one line makes the theory short and uncluttered.



Hopefully in the future the simplifier will support rewriting modulo commutativity directly, then we can remove `mul_comm`.

(Alternatively: the rule can be generalized to a `division_ring`. This requires a `calc` proof, as one part needs `pow_succ` and the other one `pow_succ'`. So it isn't a single `simp` call anymore.)